File: oving4/src/java/oving4/Oving4.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package oving4;

import javax.ws.rs.core.Context;
import javax.ws.rs.core.UriInfo;
import javax.ws.rs.PathParam;
import javax.ws.rs.Consumes;
import javax.ws.rs.Produces;
import javax.ws.rs.GET;
import javax.ws.rs.POST;
import javax.ws.rs.Path;
import javax.ws.rs.PUT;
import javax.ws.rs.core.MediaType;

/**
 * REST Web Service
 *
 * @author OddErik
 */
@Path("/oving4/")
public class Oving4 {

    @Context
    private UriInfo context;
    private static String msg = "Test";

    @GET
    @Produces(MediaType.TEXT_PLAIN)
    public String getText() {
                
        return msg;
    }

    @POST
    @Consumes(MediaType.TEXT_PLAIN)
    public void setText(String content) {
        msg = content;
    }
}

File: oving4/web/oving4.js
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
var posH = 0;
var posW = 0;

$(document).ready(function() {
    $('#oving4').mouseover(function() {
        newPos();
        $('#oving4').animate({
            left: posH,
            top: posW 
        }),1000;
    });
    
    $('#oving4').click(function() {
        $('#oving4').stop();
        $('#oving4').fadeOut();
    });
});

function newPos() {
    posH = Math.random() * window.innerHeight;
    posW = Math.random() * window.innerWidth;
}